### PR TITLE
add method to repository and service for user

### LINKED ISSUE #11: [User] Create use with role admin, user and CRUD functionality

### PR DESCRIPTION
**Pull request by QuyDang1108**

### CODE REVIEW

**Review by BinhDT1707 — APPROVED** [review]

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -1,8 +1,68 @@
...
+
+        if (!registerRequest.getPassword().equals(registerRequest.getRePassword())) {
+            return false;
+        }
+
+        if (repository.findByEmailOrPhone(registerRequest.getEmail()) != null) {
+            return false;
+        }
+
+        if (repository.findByEmailOrPhone(registerRequest.getPhone()) != null) {
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: duplicate code

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -1,8 +1,68 @@
...
+            return false;
+        }
+
+        try {
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: why use try here

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -1,8 +1,68 @@
...
+        if (registerRequest.getEmail().isEmpty()
+                || registerRequest.getPhone().isEmpty()
+                || registerRequest.getPassword().isEmpty()
+                || registerRequest.getLastname().isEmpty()
+                || registerRequest.getFirstname().isEmpty()
+                || registerRequest.getRePassword().isEmpty()) {
+            return false;
+        }
+
+        if (!registerRequest.getPassword().equals(registerRequest.getRePassword())) {
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: consider validate re-password at front-end (thymeleaf), we do not need to check it in back-end

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -1,8 +1,68 @@
...
+@RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
+    @Value("${application.admin.default.key}")
+    private String key;
+    private final UserRepository repository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public boolean checkRegister(RegisterRequest registerRequest) {
+        if (registerRequest.getEmail().isEmpty()
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: this validations maybe check in the controller, like registerRequest, we use spring validation to check, the service is only for business logic

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -1,8 +1,68 @@
...
+        }
+
+        try {
+            User user = new User();
+            user.setEmail(registerRequest.getEmail());
+            user.setFirstname(registerRequest.getFirstname());
+            user.setPhone(registerRequest.getPhone());
+            user.setLastname(registerRequest.getLastname());
+            user.setPassword(passwordEncoder.encode(registerRequest.getPassword()));
+            user.setCreatedDate(LocalDateTime.now());
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: the created date and modified date is auto generate by Audit entity -> consider remove them.

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -37,4 +37,5 @@ application:
       username: ${ADMIN_DEFAULT_USERNAME}
       password: ${ADMIN_DEFAULT_PASSWORD}
       firstname: ${ADMIN_DEFAULT_FIRST}
-      lastname: ${ADMIN_DEFAULT_LAST}
\ No newline at end of file
+      lastname: ${ADMIN_DEFAULT_LAST}
+      key: ADMIN
```
Comment (khanhduzz) [inline src/main/resources/application.yml]: something like key / secret key is better save in .env file, which is not public on github

File: src/main/java/com/fjb/sunrise/repositories/UserRepository.java [hunk 7]
```diff
@@ -2,9 +2,14 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface UserRepository extends JpaRepository<User, Long> {
     boolean existsByUsername(String username);
 
     User findByUsername(String username);
+
+    @Query("FROM User u WHERE u.email = :username OR u.phone = :username")
+    User findByEmailOrPhone(@Param("username") String username);
```
Comment (khanhduzz) [inline src/main/java/com/fjb/sunrise/repositories/UserRepository.java:14]: consider user only 1 method to check in the same time

File: src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -1,8 +1,68 @@
...
+            user.setLastModifiedDate(LocalDateTime.now());
+
+            if (registerRequest.getPassword().startsWith(key)) {
+                user.setRole(ERole.ADMIN);
+            } else {
```
Comment (QuyDang1108) [inline src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java]: em test thử thoai :v

File: src/main/java/com/fjb/sunrise/repositories/UserRepository.java (hunk 7 above)
Comment (QuyDang1108) [reply src/main/java/com/fjb/sunrise/repositories/UserRepository.java:14]: i will use it in AuthController, i dont want to delete your method

**Review by khanhduzz — APPROVED** [review]
LGTM!

**Pull request merged**